executor/aggregate.go

```diff
@@ -298,14 +300,17 @@ func (e *HashAggExec) Close() error {
 
 // Open implements the Executor Open interface.
 func (e *HashAggExec) Open(ctx context.Context) error {
-	if err := e.baseExecutor.Open(ctx); err != nil {
-		return err
-	}
 	failpoint.Inject("mockHashAggExecBaseExecutorOpenReturnedError", func(val failpoint.Value) {
```

Comment:
The failpoint is moved before the `Open`, because after calling `Open` and then error, the children Executor is not closed.